### PR TITLE
Automate release process

### PR DESCRIPTION
**Pull request by freddydk**
Automate:
- The generation of modifying release notes with the version number (and revert)
- Creation of a release in AL-Go repo

Adds:
- Check that a successful end 2 end test was run on the same commit you are releasing

### CODE REVIEW

File: .github/workflows/Deploy.yaml [hunk 1]
```diff
@@ -28,8 +33,32 @@ defaults:
...
 
 jobs:
+  CheckEndToEnd:
+    runs-on: [ ubuntu-latest ]
+    steps:
+      - name: Check successful end 2 end tests have run
+        if: github.repository_owner == 'microsoft'
+        env:
+          GH_TOKEN: ${{ github.token }}
+          repository: ${{ github.repository }}
```
Comment (aholstrup1) [inline .github/workflows/Deploy.yaml:43]: I think GitHub already has $ENV:GITHUB_REPOSITORY for this purpose

File: .github/workflows/Deploy.yaml [hunk 2]
```diff
@@ -90,3 +119,38 @@ jobs:
...
+        run: |
+          $errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
+          $releaseNotesFile = Join-Path $env:GITHUB_WORKSPACE "RELEASENOTES.md"
+          $releaseNotes = (Get-Content -Encoding utf8 -Path $releaseNotesFile) -join "`n"
+          $lastVersion = $releaseNotes.indexof("`n## ")
+          $releaseNotes = $releaseNotes.Substring(0, $lastVersion)
+          Add-Content -encoding UTF8 -Path $env:GITHUB_ENV -Value "ReleaseNotes=$([Uri]::EscapeDataString($releaseNotes))"
+
+      - name: Create release
+        if: github.repository_owner == 'microsoft' && github.event.inputs.createRelease
```
Comment (aholstrup1) [inline .github/workflows/Deploy.yaml:134]:
```suggestion
        if: github.repository_owner == 'microsoft' && github.event.inputs.createRelease == 'true'
```

Not sure if I remember this correctly, but aren't the boolean inputs to GitHub either 'true' or 'false' (as strings)? 

For example in E2E do: 
`github.event.inputs.runUpgradeTests == 'true'`

File: .github/workflows/Deploy.yaml [hunk 3]
```diff
@@ -90,3 +119,38 @@ jobs:
             Write-Host "::Error::Error deploying repositories. The error was $($_.Exception.Message)"
             exit 1
           }
+
+      - name: Calculate Release Notes
+        if: github.repository_owner == 'microsoft'
```
Comment (aholstrup1) [inline .github/workflows/Deploy.yaml:124]: Check for github.event.inputs.createRelease - No need to run this if we aren't creating a release

**Review by aholstrup1 — APPROVED** [review]

**Review by mazhelez — APPROVED** [review]

File: .github/workflows/Deploy.yaml [hunk 4]
```diff
@@ -90,3 +122,38 @@ jobs:
...
+
+      - name: Calculate Release Notes
+        if: github.repository_owner == 'microsoft' && github.event.inputs.createRelease == 'true'
+        run: |
+          $errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
+          $releaseNotesFile = Join-Path $env:GITHUB_WORKSPACE "RELEASENOTES.md"
+          $releaseNotes = (Get-Content -Encoding utf8 -Path $releaseNotesFile) -join "`n"
+          $lastVersion = $releaseNotes.indexof("`n## ")
+          $releaseNotes = $releaseNotes.Substring(0, $lastVersion)
+          Add-Content -encoding UTF8 -Path $env:GITHUB_ENV -Value "ReleaseNotes=$([Uri]::EscapeDataString($releaseNotes))"
```
Comment (mazhelez) [inline .github/workflows/Deploy.yaml:134]: `GITHUB_ENV` has some length limitation. Could release notes get big enough to reach this limitation?

Comment (freddydk) [reply .github/workflows/Deploy.yaml:134]: I think the limit is 64Kb.
We only use the env variable for the release notes for the latest version, which isn't very big.
Today the release notes for ALL versions is 41kb.

**Review by aholstrup1 — APPROVED** [review]

**Pull request merged**